lyricsheets/fonts/fonts.py

```diff
@@ -38,7 +38,7 @@ def split_by_rendered_width(
 
         return ans
 
-    def get_length(self, text: str) -> int:
+    def get_length(self, text: str) -> float:
```

Comment:
I think there are other places in the code that assume this returns an int, can check whether changing this return to a float has any effect (e.g. generating karaoke tags with long float timings)

Reply:
Maybe a round() can be added to the property accesses in `KChar`, `KSyl` and `KLine` to round the numbers? That should remove the long float timings.

Reply:
Sure

Reply:
I tried doing that, and there's another issue with it. Since some of the values calculate their values using other values, adding `round()` to the property accesses means that there's already a loss of accuracy when calculating internal values. It's fine to return a `round()` value when the properties are accessed externally, but if that's the case, a different representation is needed internally, where the values are not rounded.

Reply:
Eh looking at it again I guess it's fine since the contract of the outer methods (in `KChar`, `KSyl`, etc.) are all supposed to return floats, so any external callers who need it to be `int`s should do their own rounding

So the situation I mentioned would technically be a bug in the effect generators and not the models, so I guess this PR is OK as is (but maybe check for if anything in `default_effect.py` needs to be changed)

Reply:
It looks ok from visual inspection of a couple of generations, so it should be all good?